File: fixtures/scale/julia/medium/src/Memory.jl
module Memory

export mem_store, mem_recall, mem_clear

const SLOT = Ref{Float64}(0.0)

function mem_store(value::Float64)
    SLOT[] = value
end

function mem_recall()
    return SLOT[]
end

function mem_clear()
    SLOT[] = 0.0
end

end

File: fixtures/scale/r/medium/core/memory.r
# Single-slot memory register (M+, MR, MC).

.calc_memory <- new.env()
.calc_memory$value <- 0

mem_store <- function(value) {
  .calc_memory$value <- value
}

mem_recall <- function() {
  .calc_memory$value
}

mem_clear <- function() {
  .calc_memory$value <- 0
}
